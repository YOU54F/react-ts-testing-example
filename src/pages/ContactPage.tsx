import * as React from 'react';

export class Contact extends React.Component<{}, {}> {
    public render() {
        return (
            <div className="contact-page-container">
                <div className="form-message">
                <p className="fade-out fade-in">An example form that doesn't submit anything</p>
                <p className="fade-out fade-in" style={{ animationDelay: "0.755s" }}>Click <a href="mailto:yousafn@gmail.com" rel="noopener noreferrer">here</a> to send me an email.</p>
                </div>
                <form className="contact-form">
                    <div className="input-container fade-out fade-in" style={{ animationDelay: "1.5s" }}>
                        <input type="text" id="Name" className="input" placeholder="Name"/>
                    </div>
                    <div className="input-container fade-out fade-in" style={{ animationDelay: "2.25s" }}>
                        <input type="email" id="Email" className="input" placeholder="Email"/>
                    </div>
                    <div className="input-container fade-out fade-in" style={{ animationDelay: "3s" }}>
                        <textarea className="input" placeholder="Message"/>
                    </div>
                    <button type="button" className="button fade-out fade-in" style={{ animationDelay: "3.75s" }}>Submit</button>
                </form>
            </div>
        )
    }
}