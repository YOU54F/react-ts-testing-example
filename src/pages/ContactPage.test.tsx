import { shallow } from 'enzyme';
import * as React from 'react';
import renderer from 'react-test-renderer';
import { Contact } from './ContactPage';

describe('Contact Page', () => {
  it('renders a name input', () => {
    expect(shallow(<Contact />).find('#Name').length).toEqual(1);
  });
  it('renders a email input', () => {
    expect(shallow(<Contact />).find('#Email').length).toEqual(1);
  });
  it('renders a message input', () => {
    expect(shallow(<Contact />).find('#Message').length).toEqual(1);
  });
  it('renders a submit button', () => {
    expect(shallow(<Contact />).find('#Submit').length).toEqual(1);
  });
  it('can render the Contact page', () => {
    const tree = renderer.create(<Contact />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
