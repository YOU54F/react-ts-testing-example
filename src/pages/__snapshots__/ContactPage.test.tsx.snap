// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Page can render the Contact page 1`] = `
<div
  className="contact-page-container"
>
  <div
    className="form-message"
  >
    <p
      className="fade-out fade-in"
    >
      An example form that doesn't submit anything
    </p>
    <p
      className="fade-out fade-in"
      style={
        Object {
          "animationDelay": "0.755s",
        }
      }
    >
      Click
       
      <a
        href="mailto:yousafn@gmail.com"
        rel="noopener noreferrer"
      >
        here
      </a>
       
      to send me an email.
    </p>
  </div>
  <form
    className="contact-form"
  >
    <div
      className="input-container fade-out fade-in"
      style={
        Object {
          "animationDelay": "1.5s",
        }
      }
    >
      <input
        className="input"
        id="Name"
        placeholder="Name"
        type="text"
      />
    </div>
    <div
      className="input-container fade-out fade-in"
      style={
        Object {
          "animationDelay": "2.25s",
        }
      }
    >
      <input
        className="input"
        id="Email"
        placeholder="Email"
        type="email"
      />
    </div>
    <div
      className="input-container fade-out fade-in"
      style={
        Object {
          "animationDelay": "3s",
        }
      }
    >
      <textarea
        className="input"
        id="Message"
        placeholder="Message"
      />
    </div>
    <button
      className="button fade-out fade-in"
      id="Submit"
      style={
        Object {
          "animationDelay": "3.75s",
        }
      }
      type="button"
    >
      Submit
    </button>
  </form>
</div>
`;
